components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -338,6 +338,27 @@ void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
   rules.push_back(CloneRule(wp_com_rule));
   brave_cookie_rules_[incognito].push_back(CloneRule(wp_com_rule));
 
+  // Add ability for google properties to send cookies to each other,
+  // especially needed for google drive playback.
+  //
+  // Partial fix for https://github.com/brave/brave-browser/issues/1122
+  const auto googleapis_com_pattern = ContentSettingsPattern::FromString(
+      "https://[*.]googleapis.com/*");
+  const auto google_com_pattern = ContentSettingsPattern::FromString(
+      "https://[*.]google.com/*");
+  const std::vector<ContentSettingsPattern> patterns = {
+      googleapis_com_pattern, google_com_pattern};
+  for (const auto &outer : patterns) {
+    for (const auto &inner : patterns) {
+      const auto a_rule = Rule(
+          outer,
```

Comment:
TBH, i dont fully understand why the same domain rules were needed.  making the exception for `[*.]google.com` to `[*.]google.com` seemed like it shouldn't have been needed (same with the above).

We can remove the above, i don't think it'll be a big deal.  It just makes the thing a bit more clumsy to express.  If you'd prefer to just have the three exceptions though:

`[*.]google.com` -> `[*.]google.com`
`[*.]google.com` -> `[*.]googleapis.com`
`[*.]googleapis.com` -> `[*.]google.com`

I can update accordingly

Reply:
Shouldn't these two exceptions resolve the issue:

```
[*.]google.com -> [*.]googleapis.com
[*.]googleapis.com -> [*.]google.com
```

Reply:
From discussion with @pes10k: 
```
there were cookies that wouldn’t be sent from (deeply nested sub domains).drive.google.com to drive.google.com
```

@iefremov @bridiver -  This looks like a separate issue being fixed by `[*.]google.com -> [*.]google.com` exception. Any thoughts why this might be happening?

More details here: https://bravesoftware.slack.com/archives/C2FQMN4AD/p1587530010205900?thread_ts=1586811883.160700&cid=C2FQMN4AD

Reply:
@jumde correct, we should only need to handle `googleapis.com` <-> `google.com`

Reply:
```[*.]google.com -> [*.]google.com``` should already be fine unless you have a site exception added for blocking 3rd-party cookies (toggled to a different setting and then back again)

Reply:
This may have to do with the other 3p cookie issues uncovered and discussed today, but i can promise that this was needed to unbreak the site, and that *.google.com domains were not sending cookies to other *.google.com w/o it as of last night

Reply:
I've retested this after https://github.com/brave/brave-core/pull/5318 and cherry picking in https://github.com/brave/brave-core/pull/5341, and the `[*.]google.com` -> `[*.]google.com` exception is still needed.

@bridiver whatever is going is going on at a level of chromium / brave-core that I dont understand or can dig into anytime soon.  What do you think if I create a follow up issue (assigned to you?) to investigate further, and add a similar todo / `¯\_(ツ)_/¯` comment in the code?

Reply:
bc it would be good to get this fix out ASAP